src/Compilers/CSharp/Portable/Syntax/SyntaxNormalizer.cs

```diff
@@ -490,6 +495,18 @@ private static bool NeedsSeparator(SyntaxToken token, SyntaxToken next)
                 return true;
             }
 
+            // No space after asterisk in function pointer.
+            if (token.IsKind(SyntaxKind.AsteriskToken) && token.Parent.IsKind(SyntaxKind.FunctionPointerType))
```

Comment:
It's a bit more complicated than this. The IDE rules I implemented are:

1. If there is no calling convention, no spaces.
2. If there is a calling convention, put a space between the asterisk and the calling convention. So, for example, `delegate* unmanaged<void>`. See the IDE spacing rules here: [https://github.com/dotnet/roslyn/blob/main/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/SpacingFormattingRule.cs#L321-L413](https://github.com/dotnet/roslyn/blob/main/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/SpacingFormattingRule.cs#L321-L413).

Reply:
Thanks @333fred. This makes it much easier.